### ISSUE TITLE
[TextField] Clear button is missing focus-ring, cannot be used in popovers

### ISSUE THREAD

**Issue by voiid** [Shopify/polaris-react#3687]
## Feature request summary

Clear button is missing focus-ring for use with the new design language.

It also cannot be used in popovers, as doing so causes the popover to be closed when clicked. 

[The popover overlay `handleClick` function](https://github.com/Shopify/polaris-react/blob/6c39f4a00d7431d275d1441230814dca04f0be07/src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx#L248) determines whether or not a click was inside the popover to decide if it should close the popover or not. The clear button would inadvertently cause this to happen because the button disappears as soon as it is clicked, causing the `isDescendent` check to fail and the popover to be closed. This change ensures that the `clearButton` remains in the DOM after being clicked.

## Rationale

Accessibility.

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.
